server/routes/induction/update/previousWorkExperienceTypesUpdateController.ts

```diff
@@ -110,22 +110,36 @@ export default class PreviousWorkExperienceTypesUpdateController extends Previou
 
     req.session.previousWorkExperienceTypesForm = undefined
 
-    const pageFlowQueue = this.buildPageFlowQueue(workExperienceTypesToShowDetailsFormFor, prisonNumber)
+    const pageFlowQueue = this.buildPageFlowQueue(
+      workExperienceTypesToShowDetailsFormFor,
+      prisonNumber,
+      this.previousPageWasCheckYourAnswers(req),
+      req.session.pageFlowQueue,
+    )
+
     req.session.pageFlowQueue = pageFlowQueue
     return res.redirect(getNextPage(pageFlowQueue))
   }
 
   buildPageFlowQueue = (
     previousWorkExperienceTypes: Array<TypeOfWorkExperienceValue>,
     prisonNumber: string,
+    fromCheckAnswers: boolean,
+    currentPageFlow?: PageFlow,
   ): PageFlow => {
-    const previousWorkExperienceTypesPageUrl = `/prisoners/${prisonNumber}/induction/previous-work-experience`
-    const pageUrls = [
-      previousWorkExperienceTypesPageUrl,
-      ...previousWorkExperienceTypes.map(
-        workType => `/prisoners/${prisonNumber}/induction/previous-work-experience/${workType.toLowerCase()}`,
-      ),
-    ]
+    const nextPages = previousWorkExperienceTypes.map(
+      workType => `/prisoners/${prisonNumber}/induction/previous-work-experience/${workType.toLowerCase()}`,
+    )
+
+    // If not coming from check answers and on an existing page flow, append to the current flow
+    if (currentPageFlow && !fromCheckAnswers) {
+      return appendPagesFromCurrentPage(currentPageFlow, nextPages)
+    }
```

Comment:
If the user selects "yes" from the Worked Before page, then a flow is created which redirects them to this page before redirecting back to check your answers.

However, when the flow is created on the previous controller we don't know what work experience type will be selected on this page, so can't know the complete flow. This checks if there is an existing flow and if there is injects the needed work experience details page into the existing flow from the current page.